evalml/pipelines/components/estimators/classifiers/xgboost_classifier.py

```diff
@@ -42,8 +46,15 @@ def __init__(self, eta=0.1, max_depth=6, min_child_weight=1, n_estimators=100, r
                          random_state=random_state)
 
     def fit(self, X, y=None):
+        X = _convert_to_woodwork_structure(X)
```

Comment:
Similar to LGBM, updating to just call implementation, because Estimator will call `self.input_feature_names = list(X.columns)` but since we've changed the feature names in `_encode_categories` (since XGBoost isn't compatible with weird names), this would override the original names.